### ISSUE TITLE
Allow empty version string in nuget package dependencies

### ISSUE THREAD

**Issue by droyad** [OctopusDeploy/Issues#2813]
Depends on #2809, as it would rely on modifying the NuGet library. By itself I don't think this issue warrants forking NuGet. 

http://help.octopusdeploy.com/discussions/problems/47742-nuget-packages-deployment-bug-is-not-a-valid-version-string-for-package-dependencies

**Comment by michaelnoonan** [comment]
I've spent some time investigating this as part of solving #2809. Looping you in @kowalgta because of https://github.com/OctopusDeploy/Issues/issues/2785.

After spending time to fork https://github.com/OctopusDeploy/NuGet.Client/pulls/1 to preserve versions as-is - I'm hesitant to customize NuGet much further without a really strong case.

In this situation, the new versions of `nuget.exe` (and OctoPack and `octo.exe`) will prevent you from building a package with invalid `<dependency>` elements. I think this is good because it will prevent people building what is **now** considered by NuGet to be an "invalid" nuspec file.

I'm hesitant to change the validation https://github.com/OctopusDeploy/NuGet.Client/blob/octopack/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs#L289 because it would cause a different interpretation of the `.nuspec` file on write versus read. (Though we could argue this is OK to provide legacy support)

Workaround: In this case I would recommend re-packing the offending package files as a workaround - hopefully there aren't hundreds of "invalid" packages. If using the built-in feed in Octopus, it would be reasonably easy to write a script to unpack, remove the `<dependency>` elements, and repack them. Octopus will then be able to read the metadata and deploy the packages.

I'm going to close this issue for now because I don't feel comfortable at taking on the additional risk and management of further customising NuGet - but feel free to come back and continue the conversation if you think I've made a bad decision.

https://octopus.com/company/roadmap/why-was-this-closed
